app/models/federated_protocols/saml.rb

```diff
@@ -14,12 +17,20 @@ def ial
       if ialmax_requested_with_authn_context_comparison?
         ::Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
       else
-        request.requested_ial_authn_context || default_authn_context
+        requested_ial_authn_context || default_ial_authn_context
+      end
+    end
+
+    def requested_ial_authn_context
+      request.requested_authn_contexts.find do |classref|
+        IAL_PREFIX.match?(classref)
       end
     end
 
     def aal
-      request.requested_aal_authn_context
+      request.requested_authn_contexts.find do |classref|
+        AAL_PREFIX.match?(classref)
+      end
```

Comment:
note to other reviewers: this code is a lift and shift from the saml_idp gem (see: [PR#117: Remove ial/aal concerns](https://github.com/18F/saml_idp/pull/117/files#diff-ae34a618d7fe12815722f1afe4f0717ddf02f566c9c92597cacb44c4150d73dcL93-L104)) and will be replaced in the next sprint with a long term solution.